### PR TITLE
Update to Dotnet 8.0

### PR DESCRIPTION
**Pull request by samiz-stripe**
Updates to Dotnet 8.0

## Context

- Dotnet 5.0 is currently unsupported and does not work on Apple silicon.
- The bug squash interview for C# was previously updated to 7.0 but the interview prep is currently out of sync, causing issues for candidates: https://github.com/stripe-interview/refit
- The bug squash instructions will be updated to reference the new 8.0 requirement following this PR: https://confluence.corp.stripe.com/pages/viewpage.action?pageId=74826403

## This PR

- Updates the project to dotnet 8.0
- Bumps up `Microsoft.NET.Test.Sdk` to `17.8.0`
- Bumps up `xunit` to `2.6.6`
- Bumps up `xunit.runner.visualstudio` to `2.5.6`
- Bumps up `Newtonsoft.Json` to `13.0.3`
- Sets the minimum Visual Studio version to `17.8` as this is the minimum version required for dotnet 8.0

## Test

Build and Test output:

```
02:55:44 samiz@st-samiz2 csharp-interview-prep ±|samiz-dotnet-7 ✗|→ dotnet restore
  Determining projects to restore...
  Restored /Users/samiz/stripe/csharp-interview-prep/lib/lib.csproj (in 4.69 sec).
  Restored /Users/samiz/stripe/csharp-interview-prep/tests/tests.csproj (in 6.28 sec).
02:56:02 samiz@st-samiz2 csharp-interview-prep ±|samiz-dotnet-7 ✗|→ dotnet clean
MSBuild version 17.8.3+195e7f5a3 for .NET
Build started 1/19/2024 2:56:06 PM.
     1>Project "/Users/samiz/stripe/csharp-interview-prep/csharp-interview-prep.sln" on node 1 (Clean target(s)).
     1>ValidateSolutionConfiguration:
         Building solution configuration "Debug|Any CPU".
     1>Done Building Project "/Users/samiz/stripe/csharp-interview-prep/csharp-interview-prep.sln" (Clean target(s)).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.40
02:56:07 samiz@st-samiz2 csharp-interview-prep ±|samiz-dotnet-7 ✗|→ dotnet test
  Determining projects to restore...
  All projects are up-to-date for restore.
  lib -> /Users/samiz/stripe/csharp-interview-prep/lib/bin/Debug/net8.0/lib.dll
  tests -> /Users/samiz/stripe/csharp-interview-prep/tests/bin/Debug/net8.0/tests.dll
Test run for /Users/samiz/stripe/csharp-interview-prep/tests/bin/Debug/net8.0/tests.dll (.NETCoreApp,Version=v8.0)
Microsoft (R) Test Execution Command Line Tool Version 17.8.0 (arm64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:00.10]     tests.UnitTest1.FailingTest [FAIL]
  Failed tests.UnitTest1.FailingTest [< 1 ms]
  Error Message:
   Assert.True() Failure
Expected: True
Actual:   False
  Stack Trace:
     at tests.UnitTest1.FailingTest() in /Users/samiz/stripe/csharp-interview-prep/tests/UnitTest1.cs:line 26
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)

Failed!  - Failed:     1, Passed:     1, Skipped:     0, Total:     2, Duration: 7 ms - tests.dll (net8.0)
02:56:25 samiz@st-samiz2 csharp-interview-prep ±|samiz-dotnet-7 ✗|→ dotnet run
Couldn't find a project to run. Ensure a project exists in /Users/samiz/stripe/csharp-interview-prep, or pass the path to the project using --project.
02:56:44 samiz@st-samiz2 csharp-interview-prep ±|samiz-dotnet-7 ✗|→ dotnet run --project lib
Hello world!
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,9 +14,9 @@ setup.
 
 # Getting Ready
 
-If you have not already, please install [.NET 5.0](https://dotnet.microsoft.com/en-us/download/dotnet/5.0)
```
Comment (georgevreilly-stripe) [inline README.md:17]: .NET 7 is a [Standard Term Support release](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core) that ends support in four months, while .NET 8 is good until Nov 2026.
We should update both repos to .NET 8.

File: README.md [hunk 2]
```diff
@@ -47,7 +47,7 @@ This project uses xUnit for implementing and executing tests.
 
 # About This Project:
 
-## I prefer .Net Framework 6/nUnit/Bespoke Powershell Scripts, do I need to use this?
+## I prefer nUnit/Bespoke Powershell Scripts, do I need to use this?
 
 If you prefer different configuration files, that's awesome.
 Some interview questions may ask to implement something from nothing,
```
Comment (georgevreilly-stripe) [inline README.md:53]: "may ask you"

**Review by georgevreilly-stripe — CHANGES_REQUESTED** [review]
I think we should update both repos to .NET 8, since it's supported for another 34 months.

File: README.md (hunk 1 above)
Comment (samiz-stripe) [reply README.md:17]: Bumped it up to 8.0 and updated all dependencies

File: README.md (hunk 2 above)
Comment (samiz-stripe) [reply README.md]: fixed

**Review by georgevreilly-stripe — APPROVED** [review]
LGTM

**Pull request merged**